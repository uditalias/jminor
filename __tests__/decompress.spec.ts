'use strict';

import { createDictionary, decompress } from "../lib";

const data = require("./__mocks__/data.json");
const compressed = require("./__mocks__/compressed.json");

describe('decompress', () => {

    let dictionary;
    beforeEach(() => {
        dictionary = createDictionary().fromJSON(data);
    });

    it('should decompress data', () => {

        expect(decompress(compressed, dictionary)).toMatchSnapshot();

    });
});