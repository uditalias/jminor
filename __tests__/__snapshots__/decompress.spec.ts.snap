// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decompress should decompress data 1`] = `
Object {
  "first": 1,
  "fourteen": Array [
    Object {
      "fifteen": 0,
    },
  ],
  "second": 0,
  "seventh": Object {
    "eighth": "test",
    "eleventh": Array [],
    "tenth": Object {},
  },
  "sixth": null,
  "third": Array [
    Object {
      "fifth": "",
      "fourth": 4,
    },
    Object {
      "fifth": false,
      "fourth": true,
    },
  ],
  "twelve": Object {
    "thirteen": false,
  },
}
`;
